src/rt/sched/behaviour.h

```diff
@@ -6,6 +6,8 @@
 
 namespace verona::rt
 {
+  static thread_local bool behaviour_yielded;
```

Comment:
I wonder reading the code if `rerun` would be a better name for the concept you have introduced.  I.e. `behaviour_rerun` etc.

This is really saying we want to rerun this behaviour once it has completed.

Or perhaps even  `rerun_atomic`.

Yield implies it will stop it instantly.